janus_core/geom_opt.py

```diff
@@ -60,35 +69,56 @@ def optimize(
     dyn_kwargs = dyn_kwargs if dyn_kwargs else {}
     filter_kwargs = filter_kwargs if filter_kwargs else {}
     opt_kwargs = opt_kwargs if opt_kwargs else {}
-    struct_kwargs = struct_kwargs if struct_kwargs else {}
+    write_kwargs = write_kwargs if write_kwargs else {}
     traj_kwargs = traj_kwargs if traj_kwargs else {}
+    log_kwargs = log_kwargs if log_kwargs else {}
 
-    if struct_kwargs and "filename" not in struct_kwargs:
-        raise ValueError("'filename' must be included in struct_kwargs")
+    if write_kwargs and "filename" not in write_kwargs:
+        raise ValueError("'filename' must be included in `write_kwargs`")
```

Comment:
```suggestion
    if not write_kwargs.get("filename", None):
        raise ValueError("'filename' must be included in `write_kwargs`")
```

Reply:
Probably want a better check honestly. What is the necessity for a filename?
`not Path(filename).exists()`?

Reply:
What are you wanting it to check that it doesn't currently?

The filename option is admittedly a bit convoluted.

Personally, I'm not a fan of default file names where possible, but I think we decided we'd rather have one.

How it works currently is that if you don't pass `write_kwargs` at all, it'll use the default filename specified below (using `atoms.get_chemical_formula`), but if you're already specifying `write_kwargs`, it requires that to include `filename`.

Reply:
Changed it now so it'll always allow the default. I'm still not sure there's a perfect solution, but at least its consistent.